src/libs/sfdk/vboxvirtualmachine.cpp

```diff
@@ -429,6 +428,19 @@ void VBoxVirtualMachinePrivate::setVideoMode(const QSize &size, int depth,
         setExtraData(key, value, Sdk::instance(), [=](bool ok) { *allOk &= ok; });
     };
 
+    fetchExtraData("GUI/LastNormalWindowPosition", Sdk::instance(), [=, batch = composer.batch()](QString data, bool ok) {
+        if (!ok)
+            return;
+        QTC_ASSERT(data.startsWith("Value:"), return);
+        data = data.section(':',1,1).trimmed();
+        QStringList splitData = data.split(',');
+        QTC_ASSERT(splitData.size() > 1, return);
+        QString lastPosition = splitData[0] + ',' + splitData[1] + ','
+                + QString::number(size.width()) + ',' + QString::number(size.height());
+        BatchComposer composer = BatchComposer::extendBatch(batch);
+        enqueue("GUI/LastNormalWindowPosition", lastPosition);
+    });
+
```

Comment:
This pair of `fetchExtraData` and (optional) `setExtraData` makes a batch and so we need to be explicit about it or the `setExtraData` would get enqueued after the checkpoint below.

```
{
    BatchComposer composer = BatchComposer::createBatch("setLastNormalWindowPosition");
    
    fetchExtraData("GUI/LastNormalWindowPosition", Sdk::instance(),
            [=, batch = composer.batch()](QString data, bool ok) {
        ... unchanged ...
    });
}
```

Compare with, e.g., https://github.com/sailfishos/sailfish-qtcreator/blob/master/src/libs/sfdk/vboxvirtualmachine.cpp#L785.

Reply:
I did look at doSetReservedPortListForwarding() earlier, but I didn't notice that there were two createBatch()es :(

Reply:
Now it seems to be working as it should :)